src/utils/string/string.test.ts

```diff
@@ -260,3 +261,9 @@ describe("getDateCopy", () => {
     );
   });
 });
+
+describe("applyStrictRegex", () => {
+  test("converts string to strict regex", () => {
+    expect(applyStrictRegex("dog")).toEqual("^dog$");
+  });
```

Comment:
We should also add some test cases to check if this regex works.
```suggestion
  test("converts string to strict regex expression", () => {
    expect(applyStrictRegex("dog")).toEqual("^dog$");
  });
  test("returned regex should have a strict match", () => {
  // run the regex against a few test cases and make sure it only matches on values we expect
  })
```

Reply:
Sure, I can include some tests that the regex work in JS but I don't think it will be exhaustive since the regular expression is being processed in our mongodb query and this test will utilize a js lib util instead. 
I will include a comment that a precondition for the param is that it should not contain any regex operators. This precondition should be satisfied by the [variant field in GroupedBuildVariant](https://github.com/evergreen-ci/evergreen/blob/main/graphql/schema.graphql#L497)